src/d3a/models/myco_matcher/pay_as_clear.py

```diff
@@ -110,92 +88,63 @@ def _clearing_point_from_supply_demand_curve(self, bids, offers):
             if len(clearing) > 0:
                 return clearing[-1].rate, clearing[-1].energy
 
-    def _perform_pay_as_clear_matching(self):
-        self.sorted_bids = self.sorting(self.bids, True)
+    def get_clearing_point(self, bids, offers, current_time):
+        self.sorted_bids = self.sort_by_energy_rate(bids, True)
+        self.sorted_offers = self.sort_by_energy_rate(offers)
+        clearing = None
 
         if len(self.sorted_bids) == 0 or len(self.sorted_offers) == 0:
             return
 
         if ConstSettings.IAASettings.PAY_AS_CLEAR_AGGREGATION_ALGORITHM == 1:
             cumulative_bids = self._accumulated_energy_per_rate(self.sorted_bids)
             cumulative_offers = self._accumulated_energy_per_rate(self.sorted_offers)
-            self.state.cumulative_bids[self.now] = cumulative_bids
-            self.state.cumulative_offers[self.now] = cumulative_offers
+            self.state.cumulative_bids = cumulative_bids
+            self.state.cumulative_offers = cumulative_offers
             ascending_rate_bids = OrderedDict(reversed(list(cumulative_bids.items())))
-            return self._clearing_point_from_supply_demand_curve(
+            clearing = self._clearing_point_from_supply_demand_curve(
                 ascending_rate_bids, cumulative_offers)
         elif ConstSettings.IAASettings.PAY_AS_CLEAR_AGGREGATION_ALGORITHM == 2:
             cumulative_bids = self._discrete_point_curve(self.sorted_bids, math.floor)
             cumulative_offers = self._discrete_point_curve(self.sorted_offers, math.ceil)
             max_rate = self._populate_market_cumulative_offer_and_bid(cumulative_bids,
                                                                       cumulative_offers)
-            return self._get_clearing_point(max_rate)
+            clearing = self._get_clearing_point(max_rate)
```

Comment:
AFAIS, `_get_clearing_point` returns a tuple. You only use the rate here, right?
I suggest to refactor it to only return the rate. Also, it would be good to adapt `_clearing_point_from_supply_demand_curve` and `_get_clearing_point` to return a single value, not a list, so you dont need to select always the first value (in line 114)

Reply:
Its used by `_create_bid_offer_matchings` where both `clearing_rate` and `clearing_energy` is used.

Reply:
Ahh, you pass it as a tuple. OK.